airbyte-cdk/python/airbyte_cdk/logger.py

```diff
@@ -101,12 +102,14 @@ def log_by_prefix(msg: str, default_level: str) -> Tuple[int, str]:
 
     return log_level, rendered_message
 
+AIRBYTE_LOGGER_LOCK = Lock()
 
 class AirbyteLogger:
     def log(self, level, message):
         log_record = AirbyteLogMessage(level=level, message=message)
         log_message = AirbyteMessage(type="LOG", log=log_record)
-        print(log_message.json(exclude_unset=True))
+        with AIRBYTE_LOGGER_LOCK:
```

Comment:
IMO better fix is to return `logging.getLogger("airbyte")` instance instead of `AirbyteLogger`.
something like this
```python
AirbyteLogger = partial(logging.getLogger, "airbyte")
```

Reply:
I agree with @keu in principle, except there are two considerations.

- This will break anyone using the `AitbyteLogger.log` method directly with the string level as the input.
![](https://user-images.githubusercontent.com/14137523/155617707-9eb49a31-9fda-46fd-8928-70138cc32b65.png)

- The logger `warn` method is deprecated and will throw a warning `DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead`. 

Otherwise all the methods currently implemented on `AirbyteLogger` will "just work".